packages/server/api/src/app/pieces/piece-sync-service.ts

```diff
@@ -106,10 +107,17 @@ async function getOrThrow({ name, version }: { name: string, version: string }):
     return response.json()
 }
 
-async function listPieces(): Promise<PieceMetadataModelSummary[]> {
+async function listPieces(platformId?: string): Promise<PieceMetadataModelSummary[]> {
+    let premiumPieces: string[] | undefined
+    if (platformId) {
+        const platform = await platformService.getOne(platformId)
+        premiumPieces = platform?.premiumPieces
+    }
+
     const queryParams = new URLSearchParams()
     queryParams.append('edition', getEdition())
     queryParams.append('release', await flagService.getCurrentRelease())
+    premiumPieces && queryParams.append('premiumPieces', premiumPieces.join(','))
```

Comment:
Don't do things like that, make it array, queryParams is smart enough to serialize array, actually that's the wrong way to serialize an array.